internal/processor/awscontainerinsightprocessor/processor.go

```diff
@@ -0,0 +1,110 @@
+package awscontainerinsightprocessor
+
+import (
+	"context"
+	"errors"
+	"os"
+	"time"
+
+	// . "github.com/aws-observability/aws-otel-collector/internal/aws/containerinsightcommon"
+	"github.com/aws-observability/aws-otel-collector/internal/processor/awscontainerinsightprocessor/stores"
+	"github.com/aws-observability/aws-otel-collector/internal/processor/awscontainerinsightprocessor/structuredlogsadapter"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+)
+
+const (
+	LogGroupNameTag  = "log_group_name"
+	LogStreamNameTag = "log_stream_name"
+)
+
+type awscontainerinsightprocessor struct {
+	logger          *zap.Logger
+	started         bool
+	stores          []stores.K8sStore
+	shutdownC       chan bool
+	TagService      bool   `toml:"tag_service"`
+	ClusterName     string `toml:"cluster_name"`
+	HostIP          string `toml:"host_ip"`
+	NodeName        string `toml:"node_name"`
+	PrefFullPodName bool   `toml:"prefer_full_pod_name"`
```

Comment:
could you comment why do we need to have `PrefFullPodName` here?

Reply:
This is an option provided by cloudwatch agent. If it is true, the full pod name (with the hash) will be printed out. See details here: https://github.com/aws/amazon-cloudwatch-agent/blob/master/plugins/processors/k8sdecorator/stores/podstore_test.go#L537-L556